### ISSUE TITLE
UWP .appinstaller which depends on Microsoft.UI.Xaml crashes when ran through App Installer

### ISSUE THREAD

**Issue by nbevans** [microsoft/WindowsAppSDK#755]
If the .appinstaller file specifies a Dependencies Package for Microsoft.UI.Xaml then when ran via App Installer it will crash. If ran via PowerShell's Add-AppxPackage using the -AppInstallerFile option, it does not crash.

App Installer does not have an issues board so there is nowhere else to report this except here.

I have tried with both 2.4 (which Xamarin Forms 5.0 references) and 2.5 (vlatest nuget). Same problem.

**Comment by jtorjo** [comment]
@nbevans I'm curious how is the .appinstaller file generated. Automatically?

**Comment by nbevans** [comment]
Of course - by Visual Studio's UWP publishing process. I have tried via Azure DevOps pipeline *and* in VS via the UI screens. Same output.

**Comment by lhak** [comment]
I think I have seen similar issues where the app installer crashes when installing an msix package with a winui dependency if there is no internet connection.

**Comment by jtorjo** [comment]
@nbevans Is your app a desktop app? Because when I publish, I get an .msix as output, no .appinstaller whatsoever.
I created the .appinstaller myself by hand

**Comment by jonwis** [comment]
For folks hitting the crash, the team has asked you to file feedback using the feedback hub, under the "Developer Platform > App deployment" area.  So:

1. Start feedback hub (Win+F)
2. Create a report of the problem using that area
3. Click that "record details" button
4. Reproduce the problem
5. Send the report

Thanks!

**Comment by chendrixson** [comment]
Thank you for the detailed repro!  We've identified the issue and are working on a fix to the Desktop App Installer app which processes the .appinstaller file.  I'll post an update once it's available.

**Comment by nbevans** [comment]
> Thank you for the detailed repro! We've identified the issue and are working on a fix to the Desktop App Installer app which processes the .appinstaller file. I'll post an update once it's available.

Thanks - there is also other critical bugs in App Installer which I reported some time ago:

https://github.com/MicrosoftDocs/msix-docs/issues/191

**Comment by chendrixson** [comment]
The main desktop app installer app should now be updated with this fix, go to the Microsoft Store and click "Get Updates" to grab it.  If you're opted into the flight with WinGet, that one will be updated in a few days.

**Comment by nbevans** [comment]
Okay thanks. How long roughly until Store has rolled that update out to the majority of machines? (Or is that a silly question?)

**Comment by nbevans** [comment]
![image](https://user-images.githubusercontent.com/542188/117341415-fba42500-ae99-11eb-8d30-a20c1ea43f04.png)

It installed this - a version from 2020 - are you sure it has been published?

**Comment by jvintzel** [comment]
What OS are you on?  The app might still be flighting to ensure quality

**Comment by nbevans** [comment]
![image](https://user-images.githubusercontent.com/542188/117412547-d00e5280-af0c-11eb-90e9-20ff42e15a16.png)

**Comment by nbevans** [comment]
Unfortunately it is still crashing when trying to open my link above.

**Comment by nbevans** [comment]
Is there any news on this? Thanks!

**Comment by nbevans** [comment]
Another new release went out?

![image](https://user-images.githubusercontent.com/542188/121181368-d704e980-c859-11eb-83c9-5c86f437bebd.png)

**Comment by nbevans** [comment]
Praise the lord - this version has the fix in it!

ms-appinstaller:?source=http://byteflagshared.blob.core.windows.net/files/panconnect.mobile.uwp.broken.appinstaller

This works now - the App Installer does not crash.

Does it contain any other fixes in this release? Could you publish a change log?

**Comment by nbevans** [comment]
![image](https://user-images.githubusercontent.com/542188/123413903-33cb0880-d5ab-11eb-8dd7-165720c86445.png)

Can Microsoft remove the requirement that users "sign in" to the Windows Store in order for critical pieces of Windows infrastructure to be automatically updated? This is major impediment to rolling out App Installer bug fixes.

**Comment by riverar** [comment]
Sign in isn't actually required, you can close that window. It's misleading I know.

**Comment by jtorjo** [comment]
@riverar Wow, that is beyond sad. The point is, while you and I may know (and I know now that you told me), basically I lose one possible customer. 

I'm thinking of adding my app to the new store when that will happen, but if I'll run into the same crap as with this one, I will give up on it forever. I'm so tired of being the unpaid alpha, beta, and gamma tester for Microsoft.

**Comment by nbevans** [comment]
@riverar 

Yeah I did try everything, including closing it. But it still didn't update to the latest App Installer. In the end I signed in using my personal "Microsoft Account" (on somebody else's machine!) and only *then* would it install the latest update of App Installer. I then signed out of my account on their machine.

Why does Windows Store allow apps like App Installer to get so out of date in the first place? It seems to install updates for other things fine and relatively promptly - but App Installer seems to get ignored.

**Comment by nbevans** [comment]
Seeing this on more and more machines where App Installer won't auto-upgrade nor manual-update until you are signed in with a "Microsoft Account" in the Microsoft Store.

**Comment by jtorjo** [comment]
@nbevans shows you how much Microsoft truly cares about us

**Comment by nbevans** [comment]
> @nbevans shows you how much Microsoft truly cares about us
 
Please let's remain constructive as it is the only way these issues will get solved.

My feeling is that App Installer has mistakenly not been marked/signed by Microsoft as "Windows infrastructure" and so it is being treated like any other "optional" third-party Store app.

I say this because I have noticed that many other Windows components (like say "Sticky Notes", "Cortana", "Your Phone", "Xbox Game Bar" etc etc) *do* receive automatic updates regardless of whether the Microsoft Store app is "signed in", signed out, signed in with a non "Microsoft Account", etc.

**Comment by jvintzel** [comment]
As Rafael mentioned sign in should not be required.  I looked at this on a clean VM without an MSA attached this morning.  The device updated correctly to the retail version of the App Installer app which is version 1.4.3161.0.  What version are you trying to update to?  If I had to suspect you have used your MSA to sign up for the Windows Package Manager Insider Program.  Those users will get an update to an in-development version of the app based on the MSA signed up for the program.

**Comment by nbevans** [comment]
@jvintzel

I was just about to post a similar conclusion. I had signed up my Microsoft Account with the Windows Package Manager Insider Program and completely forgot about it. I stumbled upon the GitHub project for that - which indeed seems to comprise active work on App Installer. In the latest release is a msixbundle for App Installer (https://github.com/microsoft/winget-cli/releases/download/v1.0.11692/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle) which is providing a viable workaround now to get this onto end-user machines. First they install this App Installer msixbundle and then they can install our app afterward.

Is it safe to be pushing out these App Installer msixbundle's to our end-users? I assume so?
